admin/ryte/class-ryte.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin
+ */
+
+/**
+ * Handles the request for getting the Ryte status.
+ */
+class WPSEO_Ryte implements WPSEO_WordPress_Integration {
+
+	/**
+	 * Is the request started by pressing the fetch button.
+	 *
+	 * @var boolean
+	 */
+	private $is_manual_request = false;
+
+	/**
+	 * Constructs the object.
+	 */
+	public function __construct() {
+		$this->catch_redo_listener();
+	}
+
+	/**
+	 * Sets up the hooks.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		if ( ! self::is_active() ) {
+			return;
+		}
+
+		// Adds weekly schedule to the cron job schedules.
+		add_filter( 'cron_schedules', [ $this, 'add_weekly_schedule' ] );
+
+		// Sets the action for the Ryte fetch cron job.
+		add_action( 'wpseo_ryte_fetch', [ $this, 'fetch_from_ryte' ] );
+	}
+
+	/**
+	 * Determines if we can use the functionality.
+	 *
+	 * @return bool True if this functionality can be used.
+	 */
+	public static function is_active() {
+		if ( wp_doing_ajax() ) {
+			return false;
+		}
+
+		if ( ! WPSEO_Options::get( 'ryte_indexability' ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Hooks to run on plugin activation.
+	 */
+	public function activate_hooks() {
+		if ( $this->get_option()->is_enabled() ) {
+			$this->schedule_cron();
```

Comment:
I can't see the cron job added. Using WP-CLI there's no `wpseo_ryte_fetch` job. Command used to check after SSH/ing into VVV:

`wp cron event list --fields=hook,next_run,schedule `

Seems to me it's a pre-existing issue. WP code returns here:

https://github.com/WordPress/WordPress/blob/d784ebbcb44f10a586fed3f8b3b8001c4e958077/wp-includes/cron.php#L196-L198

it appears there's no `weekly` schedule added yet when we attempt to add the cron job so the job is not added. Investigating.